smt.py

```diff
@@ -102,6 +105,16 @@ class CmdComment(NamedTuple):
 Cmd = CmdDeclareFun | CmdDefineFun | CmdAssert | CmdCheckSat | CmdComment | CmdSetLogic
 
 
+class DeclareFunModel(NamedTuple):
+    decl_fn: CmdDeclareFun
+    expr: source.ExprT[assume_prove.VarName]
+
+
+class Model(NamedTuple):
+    sat_unsats: Sequence[CheckSatResult]
+    decl_models: Sequence[DeclareFunModel]
+
```

Comment:
Technically that's not a `Model` right? I think we should call it something else, maybe `Responses`? I think it'd be a good idea to follow SMT-LIB2 spec's wording and structure as much as possible here.

So like we could have something like:

```python
class CheckSatResponse(Enum):
     UNSAT = SMTLIB('unsat')
     SAT = SMTLIB('sat')
     UNKNOWN = SMTLIB('unknown')
     

GetModelResponse = Sequence[ModelResponse]

Response = CheckSatResponse | EchoResponse | ... | GetModelResponse | ... | GetValueResponse
Responses = Sequence[Response]
```

(section 3.9.1 of the spec)

Reply:
done